client/app/bundles/course/assessment/submission/components/Answer.tsx

```diff
@@ -0,0 +1,245 @@
+  const { question, answerId, readOnly, isSavingAnswer, uploadFiles } = props;
+  return (
+    <FileUploadAnswer
+      key={`question_${question.id}`}
+      {...{
+        question,
+        readOnly,
+        answerId,
+        isSavingAnswer,
+        uploadFiles,
+      }}
+    />
+  );
+};
+
+const Scribing = (props: ScribingAnswerProps): JSX.Element => {
+  const { question, readOnly, answerId } = props;
+  return (
+    <ScribingView
+      key={`question_${question.id}`}
+      answerId={answerId}
+      readOnly={readOnly}
+      scribing={question}
+    />
+  );
+};
+
+const VoiceResponse = (props: VoiceResponseAnswerProps): JSX.Element => {
+  const { question, answerId, readOnly, saveAnswerAndUpdateClientVersion } =
+    props;
+  return (
+    <VoiceResponseAnswer
+      key={`question_${question.id}`}
+      answerId={answerId}
+      question={question}
+      readOnly={readOnly}
+      saveAnswerAndUpdateClientVersion={saveAnswerAndUpdateClientVersion}
+    />
+  );
+};
+
+const ForumPostResponse = (props: ForumResponseAnswerProps): JSX.Element => {
+  const { question, answerId, readOnly, saveAnswerAndUpdateClientVersion } =
+    props;
+  return (
+    <ForumPostResponseAnswer
+      key={`question_${question.id}`}
+      answerId={answerId}
+      question={question}
+      readOnly={readOnly}
+      saveAnswerAndUpdateClientVersion={saveAnswerAndUpdateClientVersion}
+    />
+  );
+};
+
+export type AnswerType =
+  | 'MultipleChoice'
+  | 'MultipleResponse'
+  | 'Programming'
+  | 'TextResponse'
+  | 'Comprehension'
+  | 'FileUpload'
+  | 'Scribing'
+  | 'VoiceResponse'
+  | 'ForumPostResponse';
+
+export const AnswerMapper = {
+  MultipleChoice: (props: McqMrqAnswerProps): JSX.Element =>
+    MultipleChoice(props),
+  MultipleResponse: (props: McqMrqAnswerProps): JSX.Element =>
+    MultipleResponse(props),
+  Programming: (props: ProgrammingAnswerProps): JSX.Element =>
+    Programming(props),
+  TextResponse: (props: TextAnswerProps): JSX.Element => TextResponse(props),
+  Comprehension: (props: TextAnswerProps): JSX.Element => TextResponse(props),
+  FileUpload: (props: FileUploadAnswerProps): JSX.Element => FileUpload(props),
+  Scribing: (props: ScribingAnswerProps): JSX.Element => Scribing(props),
+  VoiceResponse: (props: VoiceResponseAnswerProps): JSX.Element =>
+    VoiceResponse(props),
+  ForumPostResponse: (props: ForumResponseAnswerProps): JSX.Element =>
+    ForumPostResponse(props),
+};
+
+const Answer = (props: Props): JSX.Element => {
+  const Component = AnswerMapper[props.type];
+  const { t } = useTranslation();
+
+  if (props.question.viewHistory) {
+    return <PastAnswers question={props.question} />;
+  }
+
+  if (!Component) {
+    return (
+      <Card style={{ backgroundColor: yellow[100] }}>
+        <CardContent>{t(translations.rendererNotImplemented)}</CardContent>
+      </Card>
+    );
+  }
+
+  return Component(props.answerProps as any);
```

Comment:
Add a comment why any is used here. And disable any warning only for this line, not the entire file.